src/targets/symbolCollector.ts

```diff
@@ -0,0 +1,110 @@
+import { stringToRegexp } from '../utils/filters';
+import { BaseArtifactProvider } from '../artifact_providers/base';
+import { TargetConfig } from '../schemas/project_config';
+import { ConfigurationError } from '../utils/errors';
+import { BaseTarget } from './base';
+import { withTempDir } from '../utils/files';
+import { promises as fsPromises } from 'fs';
+import { checkExecutableIsPresent, spawnProcess } from '../utils/system';
+import { join } from 'path';
+
+const DEFAULT_SYM_COLLECTOR_SERVER_ENDPOINT =
+  'https://symbol-collector.services.sentry.io/';
+/**
+ * Name of the binary of the symbol collector.
+ * Must be available in the path.
+ */
+export const SYM_COLLECTOR_BIN_NAME = 'SymbolCollector.Console';
+
+/** Config options for the "symbol-collector" target. */
+interface SymbolCollectorTargetConfig {
+  /** Server endpoint to upload symbols. */
+  serverEndpoint: string;
+  /** batch-type of the symbols to be uploaded. */
+  batchType: string;
+  /** Prefix of the bundle ID to be uploaded. */
+  bundleIdPrefix: string;
+}
+
+export class SymbolCollector extends BaseTarget {
+  /** Target name */
+  public readonly name: string = 'symbol-collector';
+  /** Target options */
+  public readonly symbolCollectorConfig: SymbolCollectorTargetConfig;
```

Comment:
This seems like this should be private

Reply:
Let's also then not make tests directly assert on this object as well, feels fragile to me.

Reply:
All the configs in the targets are public, and that's the reason why this is too. If we're going to make them private, I think we should make them all private and not have some that are public and some others that aren't.

> Let's also then not make tests directly assert on this object as well, feels fragile to me.

Any suggestions to test whether the target has the required config parameters? By looking at the codebase, I don't think having getters is better; if we're changing the conventions, we shouldn't do it for a single target. Wdyt?

Reply:
Ok we should match the conventions - disregard my thoughts then. Maybe something to keep in mind if we decide to refactor the targets in the future.

In terms of the tests, I always prefer checking if the behaviour enabled by different config options is working rather than asserting on the config option itself. Asserting on resulting behaviour produces stronger tests and also matches real life usage much better - don’t find much value from just expecting the config property to be a certain value.

Reply:
The behavior is either throwing an error (because the config wasn't correct) or successfully running the `publish` step. Not throwing an error is covered (there are tests with incorrect config expecting it to throw errors), and the `publish` step has its own tests too. Is there something that I'm missing?

Reply:
As much as I agree with @AbhiPrasad in a general sense, I think for this specific use case, this is fine. As long as the config is readonly, it is good enough for now.